roles/manage_transit_gateway/tasks/delete.yaml

```diff
@@ -0,0 +1,54 @@
+---
+- name: Run delete_tmanage_transit_gateway_gw_attach_vpc role
+  module_defaults:
+    group/aws: "{{ aws_role_credentials }}"
+  block:
+    - name: List all the transit gateway attachments
+      community.aws.ec2_transit_gateway_manage_transit_gateway_info:
+      register: manage_transit_gateway_info
+
+    - name: Get the transit gateway with the given description
+      ansible.builtin.set_fact:
+        manage_transit_gateway_gw: "{{ item }}"
+      when:
+        - item.description == "{{ transit_gateway.description }}"
+        - item.state == "available"
+      loop: "{{ manage_transit_gateway_info.transit_gateways }}"
```

Comment:
```suggestion
        manage_transit_gateway_gw: "{{ manage_transit_gateway_info.transit_gateways | selectattr('state', 'equalto', 'available') | selectattr('description', 'equalto', transit_gateway.description) | list | first }}"
```

Reply:
@abikouo Just curious - what is the advantage of using this line instead of when + loop, except for the decrease in the number of lines of code?

Reply:
I am not sure there is another advantage than the number of lines of code, you can use both